components/pd_client/src/util.rs

```diff
@@ -51,23 +55,23 @@ pub struct HeartbeatReceiver {
 }
 
 impl Stream for HeartbeatReceiver {
-    type Item = RegionHeartbeatResponse;
-    type Error = Error;
+    type Item = Result<RegionHeartbeatResponse>;
 
-    fn poll(&mut self) -> Poll<Option<Self::Item>, Error> {
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
         loop {
             if let Some(ref mut receiver) = self.receiver {
-                match receiver.poll() {
-                    Ok(Async::Ready(Some(item))) => return Ok(Async::Ready(Some(item))),
-                    Ok(Async::NotReady) => return Ok(Async::NotReady),
+                match Pin::new(&mut receiver.compat()).poll_next(cx) {
+                    Poll::Ready(Some(Ok(item))) => return Poll::Ready(Some(Ok(item))),
+                    Poll::Pending => return Poll::Pending,
                     // If it's None or there's error, we need to update receiver.
                     _ => {}
                 }
             }
 
             self.receiver.take();
 
-            let mut inner = self.inner.wl();
+            let inner_lock = self.inner.clone();
```

Comment:
One way to avoid this clone is to explicitly drop `inner` before `self.receiver = receiver;`